lib/cisco_node_utils/vxlan_vtep_vni.rb

```diff
@@ -63,8 +71,34 @@ def set_args_keys(hash={})
     end
     # rubocop:enable Style/AccessorMethodNamefor
 
+    def create_with_associate_vrf?
+      !@set_args[:vrf].eql?('')
+    end
+
+    def destroy_existing(key)
+      getargs = { name: @name, vni: @vni, state: '' }
+      # rubocop:disable Style/GuardClause
+      if config_get('vxlan_vtep', key, getargs)
+        key.eql?('vni_with_vrf') ? vrf = 'associate-vrf' : vrf = ''
+        getargs[:vrf] = vrf
+        getargs[:state] = 'no'
+        config_set('vxlan_vtep', 'vni', getargs)
+      end
+      # rubocop:enable Style/GuardClause
+    end
```

Comment:
Because it was late and I was tired and it annoyed me :)  I'll update with your suggestion.